src/main/java/org/ethereum/beacon/discovery/pipeline/handler/HandshakeMessagePacketHandler.java

```diff
@@ -130,7 +134,7 @@ public void handle(Envelope envelope) {
 
       session.setState(AUTHENTICATED);
       envelope.remove(Field.PACKET_HANDSHAKE);
-      enr.ifPresent(session::onNodeRecordReceived);
+      enr.filter(addressAccessPolicy::allow).ifPresent(session::onNodeRecordReceived);
```

Comment:
I'm wondering if we should fail the handshake completely, so apply the filter as early as possible..

Reply:
Yeah seems reasonable.